Don't allow window zoom when CanResize=false and custom window chrome

grokys opened the pull request:
## What does the pull request do?

Following on from #10575, disable window zoom/resize when `CanResize == false` in two other scenarios when `ExtendClientAreaToDecorationsHint == true`:

- On macOS: double clicking the titlebar area to zoom was handled in `WindowImpl` but didn't check the `CanResize` state
- On Windows: The maximize button in `CaptionButtons` wasn't being disabled when `CanResize == false`

Moved the macOS-specific integration test for this case to run also on Windows, and extended the test to also test when `ExtendClientAreaToDecorationsHint `.

In addition fixed a 🤦 in `ControlAutomationPeer` that meant that setting `AutomationProperties.Name` wasn't working (which was needed for the integration test).

grokys committed:
Don't allow window zoom when CanResize=false.

grokys committed:
Disable maximize button when CanResize=false.

grokys committed:
Fix ControlAutomationPeer.GetName.

:facepalm:

grokys committed:
Run check for disabled maximize button on Windows.

And also check with client-side decorations.

grokys commented on src/Avalonia.Themes.Fluent/Controls/CaptionButtons.xaml:
```diff
@@ -48,15 +48,17 @@
           </Button>
           <Button x:Name="PART_RestoreButton"
-                  Theme="{StaticResource FluentCaptionButton}">
+                  Theme="{StaticResource FluentCaptionButton}"
```
Hmm strange, I thought I also changed the default theme. Will put that back in.

grokys replied:
Done.

grokys committed:
Update simple theme caption buttons.

maxkatz6 committed:
Merge branch 'master' into fixes/custom-chrome-canresize-maximize

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0032628-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

maxkatz6 commented:
@grokys seems like windows integration tests are failing

maxkatz6 committed:
Merge branch 'master' into fixes/custom-chrome-canresize-maximize

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0032824-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

grokys committed:
Remove debug code.

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0032832-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

grokys commented:
> seems like windows integration tests are failing

Strange, looks like it's unable to find the maximize button on the CI machine, though it can find it here.

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0032837-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

grokys committed:
Get window chrome maximize button by index.

The real Windows Maximize button has no automation ID and is localized. Get it by index instead of by name. The Avalonia one isn't localized so we can get it by name.

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0032841-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

maxkatz6 committed:
Merge branch 'master' into fixes/custom-chrome-canresize-maximize

maxkatz6 committed:
Merge branch 'master' into fixes/custom-chrome-canresize-maximize

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0032953-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

grokys committed:
Merge branch 'master' into fixes/custom-chrome-canresize-maximize

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0032975-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

grokys committed:
Merge branch 'master' into fixes/custom-chrome-canresize-maximize

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0033066-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

maxkatz6 committed:
Merge branch 'master' into fixes/custom-chrome-canresize-maximize

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0033074-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]
